### ISSUE TITLE
Squirrel.SetPrintFunc in Sq.Managed.cs causes execution engine exceptions

### ISSUE THREAD

**Issue by DarkKitsune** [DaZombieKiller/SqNET#3]
The method Squirrel.SetPrintFunc creates new lambda functions before passing them to Unmanaged.SetPrintFunc but does not store them in any field or property or something so they sometimes get GCed which consistently caused an ExecutionEngineException to be thrown when I tried to print a little while after my Squirrel VM had been set up.
![Annotation 2019-07-07 005552](https://user-images.githubusercontent.com/10158953/60765607-013c4400-a052-11e9-9900-fc37e0c2b194.png)

Edit: It appears to affect other methods that use lambdas such as Squirrel.​NewClosure, as well.

**Comment by DaZombieKiller** [comment]
Yeah, the current state of this binding is rather flawed and I have intentions to rewrite it at some point once I have more time. In the meantime I'm more than willing to accept pull requests.
